src/core/server/http/router/request.test.ts

```diff
@@ -66,6 +66,24 @@ describe('KibanaRequest', () => {
     });
   });
 
+  describe('isSytemApi property', () => {
+    it('is false when no kbn-system-api header is set', () => {
+      const request = httpServerMock.createRawRequest({
+        headers: { custom: 'one' },
+      });
+      const kibanaRequest = KibanaRequest.from(request);
+      expect(kibanaRequest.isSystemApi).toBe(false);
+    });
+
+    it('is true when kbn-system-api header is set', () => {
+      const request = httpServerMock.createRawRequest({
+        headers: { custom: 'one', 'kbn-system-api': 'true' },
+      });
+      const kibanaRequest = KibanaRequest.from(request);
+      expect(kibanaRequest.isSystemApi).toBe(true);
+    });
```

Comment:
NIT: maybe add a `it('is false when kbn-system-api header is set to false')` (even if the kibana client is sending undefined in that case, we might want to ensure that `false` is properly interpreted?)